### PR TITLE
Credentials

### PR DESCRIPTION
**Pull request by rawagner**
Screenshot from 2020-05-27 12-47-52

### CODE REVIEW

File: src/api/types.ts [hunk 1]
```diff
@@ -432,3 +432,9 @@ export interface SystemVendor {
   productName?: string;
   manufacturer?: string;
 }
+
```
Comment (jtomasek) [inline src/api/types.ts:435]: this file is autogenerated, put it somewhere else for the time being please

File: src/components/clusterDetail/ClusterCredentials.tsx [hunk 2]
```diff
@@ -1,35 +1,81 @@
...
+          <dd>{credentials.username}</dd>
           <dt>Password</dt>
-          <dd>-------------</dd>
+          <dd>
+            <ClipboardCopy
+              className="cluster-detail__password"
+              isReadOnly
+              onCopy={(event) => clipboardCopyFunc(event, credentials.password)}
+            >
+              &bull;&bull;&bull;&bull;&bull;
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterCredentials.tsx:60]: It's a bit pity that  ClipboardCopy won't allow us to pass the input type (password) it should render. :/

File: src/components/clusterDetail/ClusterDetail.tsx [hunk 3]
```diff
@@ -36,6 +37,28 @@ const installationSteps = [
...
+        const response = await getClusterCredentials(cluster.id);
+        setCredentials(response.data);
+      } catch (err) {
+        setCredentialsError(err);
+      }
+    };
+    fetch();
+  }, [cluster.id]);
+
+  React.useEffect(() => {
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterDetail.tsx:56]: We could avoid doing this by moving it into the ClusterCredentials. Is it on purpose to make the ClusterCredentials without side effects?

File: src/components/clusterDetail/ClusterCredentials.tsx (hunk 2 above)
Comment (rawagner) [reply src/components/clusterDetail/ClusterCredentials.tsx:60]: at least you cant tell how long the password is :D

File: src/components/clusterDetail/ClusterDetail.tsx (hunk 3 above)
Comment (rawagner) [reply src/components/clusterDetail/ClusterDetail.tsx:56]: we still need console URL so fetching the data in this component makes sense. useEffect without side effects is completely fine IMO

File: src/components/clusterDetail/ClusterCredentials.tsx (hunk 2 above)
Comment (rawagner) [reply src/components/clusterDetail/ClusterCredentials.tsx:60]: note: we also dont have `reveal password` button, but i would argue how useful that actually is

File: src/api/types.ts (hunk 1 above)
Comment (rawagner) [reply src/api/types.ts]: moved away for now

**Comment by rawagner** [comment]
@jtomasek done

File: src/components/clusterDetail/ClusterCredentials.tsx [hunk 4]
```diff
@@ -1,35 +1,81 @@
...
-        variant={ButtonVariant.secondary}
-        onClick={() => saveAs(getClusterFileURL(cluster.id, 'kubeconfig'))}
-      >
-        Download kubeconfig
-      </Button>
+    );
+  }
+
+  return (
+    <GridItem>
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterCredentials.tsx:69]:
```suggestion
    <GridItem span={12} lg={10} xl={6}>
```

Could solve the width problem without having to use custom css.

File: src/components/clusterDetail/ClusterDetail.css [hunk 5]
```diff
@@ -5,3 +5,7 @@
 .cluster-detail__details-list dd {
   margin-bottom: var(--pf-global--spacer--sm);
 }
+
+.cluster-detail__password {
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterDetail.css:9]: not needed with suggestion above

File: src/components/clusterDetail/ClusterCredentials.tsx (hunk 4 above)
Comment (rawagner) [reply src/components/clusterDetail/ClusterCredentials.tsx]: fixed

**Pull request merged**